dkg/sync/server.go

```diff
@@ -209,6 +214,29 @@ func (s *Server) handleStream(ctx context.Context, stream network.Stream) error
 	}
 }
 
+// validReq returns an error message and false if the request version or definition hash are invalid.
+// Else it returns true or an error.
+func (s *Server) validReq(ctx context.Context, pubkey crypto.PubKey, msg *pb.MsgSync) (string, bool, error) {
```

Comment:
we are reading `s.version` and `s.defHash` without locking mutex

Reply:
they are immutable fields, no need to lock